### PR TITLE
Bump ruby 2.6.5 --> 2.7.1 and fix deprecations

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Bump ruby 2.6.5 --> 2.7.1 and fix deprecations

#### Ticket

maintenance and support

#### Why
We are falling behind plus there are performance improvements
and a need to handle deprecations in preparation for ruby 3.0.

#### How
Bump to 2.7.1 and then identify and fix the numerous deprecations
listed below.

Note that 2.7.2 will remove deprecation warnings by default and so
bumping to 2.7.1 enables us to first easily identify and fix the
deprecations. We can then bump to 2.7.2 easily afterward.

--------

#### Bump ruby

- [X] app version
- [X] circleCI version
- [X] expire circleCI cache
- [X] Dockerfile version

#### Fix deprecations

- [X] spec/vcr_helper.rb:35: warning: deprecated Object#=~ is called on FalseClass; it always returns nil

- [X] spec/vcr_helper.rb:36: warning: deprecated Object#=~ is called on FalseClass; it always returns nil

- [X] app/services/claims/claim_actions_service.rb:6: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/button_helpers.rb: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/phase_banner_helpers.rb:8: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/link_helpers_spec.rb: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/summary_list_helpers.rb:8: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/warning_text_helpers.rb:10 warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/detail_helpers.rb:11: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/panel_helpers.rb:8: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/tag_helpers.rb:9: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] lib/govuk_component/inset_text_helpers.rb:9: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] app/form_builders/adp_form_builder.rb:31: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] app/services/claims/fee_calculator/unit_price.rb:50: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] app/services/claims/fee_calculator/graduated_price.rb:50: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] app/services/claims/external_user_claim_updater.rb:23: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] app/interfaces/api/custom_validations/date_format.rb:24: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] spec/models/stage_spec.rb:13: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] spec/models/stage_transition_spec.rb:13: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [X] app/services/maps/distance_calculator.rb:22: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the cal

#### These are other libraries/gems  - not sure we can do anything about these

- [x] factory_bot-6.1.0/lib/factory_bot/decorator.rb:16: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

- [ ] gems/capybara-3.35.3/lib/capybara/session.rb:762: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call

### CODE REVIEW

**Comment by jrmhaig** [comment]
I have fixed the deprecation warning for Factory Bot but I don't see the other outstanding ones listed in the description. I would be interesting to know how to reproduce them.

**Comment by jsugarman** [comment]
This breaks the "Duplicate expense" functionality that uses `govuk_link_button_secondary` component. However no tests break. This indicates the existing tests for components at least, are insufficient. 

1. need a feature test for "Duplicate expense" functionality, at least
2. need unit tests that handle argument passing order as this is impacted by the ruby 2.7 bump

**ruby 2.6**
` def govuk_link_button_secondary(name = nil, options = nil, **tag_options, &block)`

```
govuk_link_button_secondary('Duplicate last expense', class: 'fx-duplicate-expense')
=>
<a class="govuk-button govuk-button--secondary fx-duplicate-expense" href="<current_path>" data-module="govuk-button" draggable="false" role="button">Duplicate last expense</a>

```
**ruby 2.7**
`govuk_link_button_secondary(name = nil, options = nil, tag_options = {}, &block)`

```
govuk_link_button_secondary('Duplicate last expense', class: 'fx-duplicate-expense')
=>
<a class="govuk-button govuk-button--secondary" href="<current_path>?class=fx-duplicate-expense" data-module="govuk-button" draggable="false" role="button">Duplicate last expense</a>
```
**^^ i.e. the class is added as a query _<current_path>?class=fx-duplicate-expense_**

File: spec/factories/claim/advocate_claims.rb [hunk 1]
```diff
@@ -11,7 +11,7 @@
     # expected values are not yet set.
     # More details can be found here:
     # https://stackoverflow.com/questions/5916162/problem-with-factory-girl-association-and-after-initialize
-    initialize_with { new(attributes) }
+    initialize_with { new(**attributes) }
```
Comment (jsugarman) [inline spec/factories/claim/advocate_claims.rb:14]: 🎖️ excellent

**Review by jrmhaig — APPROVED** [review]
:+1:

**Review by rahooligan — APPROVED** [review]
